zeebe/backup/src/main/java/io/camunda/zeebe/backup/management/InProgressBackupImpl.java

```diff
@@ -176,26 +179,37 @@ public ActorFuture<Void> findSnapshotFiles() {
   @Override
   public ActorFuture<Void> findSegmentFiles() {
     final ActorFuture<Void> filesCollected = concurrencyControl.createFuture();
-    try (final var stream = Files.list(segmentsDirectory)) {
-      final var fileSet =
-          stream
-              .filter(isSegmentsFile)
-              .collect(
-                  Collectors.toMap(
-                      path -> segmentsDirectory.relativize(path).toString(), path -> path));
-
-      if (fileSet.isEmpty()) {
-        filesCollected.completeExceptionally(
-            new IllegalStateException("Segments must not be empty"));
-        return filesCollected;
+    try {
+      long maxIndex = 0L;
+      if (availableValidSnapshots != null) {
+        maxIndex =
+            availableValidSnapshots.stream()
+                .mapToLong(PersistedSnapshot::getIndex)
+                .max()
+                .orElse(0L);
       }
-
-      segmentsFileSet = new NamedFileSetImpl(fileSet);
-      filesCollected.complete(null);
-    } catch (final IOException e) {
+      final var segments = journalInfoProvider.getTailSegments(maxIndex);
+      segments.whenComplete(
+          (journalMetadata, throwable) -> {
+            if (throwable != null) {
+              filesCollected.completeExceptionally(throwable);
+            } else if (journalMetadata.segments().isEmpty()) {
+              filesCollected.completeExceptionally(
+                  new IllegalStateException("Segments must not be empty"));
+            } else {
+              final Map<String, Path> map =
+                  journalMetadata.segments().values().stream()
```

Comment:
:wrench: Considering the main consumer of the API ends up looking at only the values, do we need to return a map there? Can't we just return a `Collection<Path>`? That would also remove a bunch of other refactoring done around the signature there.

Reply:
The reasoning was that if we need to then filter the paths from [snapshot_index..commit_index] in the future it may be helpful to have a SortedMap, but the size of the map is probably negligible so it does not matter much.

Reply:
I think it's probably better to do as you suggested and decide in the future if the api needs to be changed.